### ISSUE TITLE
similarity

### ISSUE THREAD

**Issue by reza1615** [man-group/dtale#264]
finding similarity between two column strings. there is many algorithms i suggest to have these options
* Hamming distance
* Levenshtein distance
* Damerau-Leveneshtein distance
* Jaro-Winkler distance
* Jaccard index
https://github.com/luozhouyang/python-string-similarity
https://pypi.org/project/Distance/

**Comment by aschonfeld** [comment]
@reza1615 so is the idea that I have to take the set of distinct values for two string columns and pass them to each of these functions to whether they are similar or not?

**Comment by reza1615** [comment]
@aschonfeld The First one. 



e.g.

| ColumnA  | ColumnB | Distance|
| ------------- |:-------------:| -----:|
| Aschonfeld      | Uschonfeld | 1 |
| Reza      | Reza      |   0|


Also I found new libraries which are a wrapper for C language and they are extremely fast. please use the below libraries

[Jellyfish ](https://github.com/jamesturk/jellyfish)
[editdistance](https://github.com/roy-ht/editdistance)

**Comment by reza1615** [comment]
Also for applying function to rows there is a library (**swifter**) which uses multicore process and some time make it 100x faster!
If you want to use it use this sintax
``` python
def theFunction(x):
    # do some thing
    return x


import swifter
df.swifter.process_bar(False).apply(theFunction, axis=1)
```

**Comment by aschonfeld** [comment]
included in v1.18.0 (FYI: swifter is a very heavy weight package.  I really like it but I think I need to do some more work to make it an optional dependency where it will use it if you have it and not use it otherwise)
